### ISSUE TITLE
Fix parser tests

### ISSUE THREAD

**Issue by anatoliykmetyuk** [scala-subscript/subscript#16]
With new calls wrapped in "maybeVarCall", most of the tests broke.

**Comment by anatoliykmetyuk** [comment]
In fact, better test not the result of AST compilation into strings but generated ASTs themselves. Tests are much more clear that way.

**Comment by AndreVanDelft** [comment]
I don't understand what went wrong and why. 
Could you also show what the tests look like when comparing to ASTs?

**Comment by anatoliykmetyuk** [comment]
Currently, the tests check the strings the parser rewrites the code to. For example, “do a then b” rewrites to:

subscript.DSL._do_then(subscript.DSL._maybeCall("", (here: subscript.vm.model.callgraph.CallGraphTreeNode) => subscript.DSL._maybeVarCall("a")), subscript.DSL._maybeCall("", (here: subscript.vm.model.callgraph.CallGraphTreeNode) => subscript.DSL._maybeVarCall("b")))}

 Problems with this:
Hard to write the test cases. Even when helper methods are used to construct such large strings, the resulting tests look like s"$doThen($a, $b)” - requires the developer to look up the meaning of these methods to some other class. Not descriptive.
Hard to debug. Even a space out of place matters, and every time something breaks I go to http://text-compare.com/ to make a char-wise diff of the real vs expected output.
Changes that have nothing to do with the parser break its tests. For example, if a DSL method is renamed, the output string changes and tests break, though the parser itself remains correct.

To summarize, what matters in parser is the correct AST parsing, not the correct code rewriting. The correct rewriting is caught easily by the OperatorsSute tests and the core-scripts module tests.
